flink-connector-prometheus/src/main/java/org/apache/flink/connector/prometheus/sink/PrometheusSinkBuilder.java

```diff
@@ -0,0 +1,264 @@
+    }
+
+    public PrometheusSinkBuilder setHttpUserAgent(String httpUserAgent) {
+        this.httpUserAgent = httpUserAgent;
+        return this;
+    }
+
+    public PrometheusSinkBuilder setErrorHandlingBehaviourConfiguration(
+            SinkWriterErrorHandlingBehaviorConfiguration errorHandlingBehaviorConfig) {
+        this.errorHandlingBehaviorConfig = errorHandlingBehaviorConfig;
+        return this;
+    }
+
+    public PrometheusSinkBuilder setMetricGroupName(String metricGroupName) {
+        this.metricGroupName = metricGroupName;
+        return this;
+    }
+
+    private Integer getMaxBatchSizeInSamples() {
+        return maxBatchSizeInSamples;
+    }
+
+    private Integer getMaxRecordSizeInSamples() {
+        return maxRecordSizeInSamples;
+    }
+
+    public RetryConfiguration getRetryConfiguration() {
+        return retryConfiguration;
+    }
+
+    public Integer getSocketTimeoutMs() {
+        return socketTimeoutMs;
+    }
+
+    public String getHttpUserAgent() {
+        return httpUserAgent;
+    }
+
+    public SinkWriterErrorHandlingBehaviorConfiguration getErrorHandlingBehaviorConfig() {
+        return errorHandlingBehaviorConfig;
+    }
+
+    public String getMetricGroupName() {
+        return metricGroupName;
+    }
+
+    /// Disable accessing maxBatchSize, maxBatchSizeInBytes, and maxRecordSizeInBytes directly
+
+    /** Not supported. Use setMaxBatchSizeInSamples(int) instead */
+    @Override
+    public PrometheusSinkBuilder setMaxBatchSize(int maxBatchSize) {
+        throw new UnsupportedOperationException("maxBatchSize is not supported by this sink");
+    }
+
+    /** Not supported. Use setMaxBatchSizeInSamples(int) instead */
+    @Override
+    public PrometheusSinkBuilder setMaxBatchSizeInBytes(long maxBatchSizeInBytes) {
+        throw new UnsupportedOperationException(
+                "maxBatchSizeInBytes is not supported by this sink");
+    }
+
+    /** Not supported. Use setMaxRecordSizeInSamples(int) instead */
+    @Override
+    public PrometheusSinkBuilder setMaxRecordSizeInBytes(long maxRecordSizeInBytes) {
+        throw new UnsupportedOperationException(
+                "maxRecordSizeInBytes is not supported by this sink");
+    }
+
+    /** Not supported. Use getMaxBatchSizeInSamples() instead */
+    @Override
+    protected Integer getMaxBatchSize() {
+        throw new UnsupportedOperationException("maxBatchSize is not supported by this sink");
+    }
+
+    /** Not supported. Use getMaxBatchSizeInSamples() instead */
+    @Override
+    protected Long getMaxBatchSizeInBytes() {
+        throw new UnsupportedOperationException(
+                "maxRecordSizeInBytes is not supported by this sink");
+    }
+
+    /** Not supported. Use getMaxRecordSizeInSamples() instead */
+    @Override
+    protected Long getMaxRecordSizeInBytes() {
+        throw new UnsupportedOperationException(
+                "maxRecordSizeInBytes is not supported by this sink");
+    }
```

Comment:
But if someone does, because they are creating their custom extended sink and they think to reuse this, they will get to the methods exposed by `AsyncSinkBaseBuilder`, potentially causing a mess.
I think shadowing those methods is safer and harmless.